scripts/ci-entrypoint.sh

```diff
@@ -82,11 +82,12 @@ select_cluster_template() {
         export CI_VERSION="${CI_VERSION:-$(curl -sSL ${CI_VERSION_URL})}"
         export KUBERNETES_VERSION="${CI_VERSION}"
     else
-        export CLUSTER_TEMPLATE="test/ci/cluster-template-prow.yaml"
+        export CLUSTER_TEMPLATE="test/ci/cluster-template-prow-in-tree-cloud-provider.yaml"
```

Comment:
Thanks for that call-out. I wonder if something as simple as this would work.

Changes to capz:

```
$ git diff test/e2e/conformance_test.go
diff --git a/test/e2e/conformance_test.go b/test/e2e/conformance_test.go
index 5106d834..a4f15eba 100644
--- a/test/e2e/conformance_test.go
+++ b/test/e2e/conformance_test.go
@@ -104,6 +104,9 @@ var _ = Describe("Conformance Tests", func() {
 
                kubernetesVersion := e2eConfig.GetVariable(capi_e2e.KubernetesVersion)
                flavor := clusterctl.DefaultFlavor
+               if os.Getenv("IN_TREE_CLOUDPROVIDER") == "true" {
+                       flavor = "in-tree-cloud-provider"
+               }
                if isWindows(kubetestConfigFilePath) {
                        flavor = getWindowsFlavor()
                }
```

Changes to test-infra specs:

```
$ git diff config/jobs/kubernetes/sig-cloud-provider/azure/release-master.yaml
diff --git a/config/jobs/kubernetes/sig-cloud-provider/azure/release-master.yaml b/config/jobs/kubernetes/sig-cloud-provider/azure/release-master.yaml
index 4af65012d0..f383e8c056 100644
--- a/config/jobs/kubernetes/sig-cloud-provider/azure/release-master.yaml
+++ b/config/jobs/kubernetes/sig-cloud-provider/azure/release-master.yaml
@@ -322,6 +322,8 @@ periodics:
         value: "latest"
       - name: CONFORMANCE_WORKER_MACHINE_COUNT
         value: "2"
+      - name: IN_TREE_CLOUDPROVIDER
+        value: "true"
       securityContext:
         privileged: true
       resources:
```

(We would also make the same env var addition to the remaining "release-*" test-infra test specs under `config/jobs/kubernetes/sig-cloud-provider/azure/`)

Reply:
O.K., this change was made. Here's the test-infra equivalent that we'll want to land as part of this effort:

https://github.com/kubernetes/test-infra/pull/25598